src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -92,29 +82,36 @@ winrt::com_ptr<Profile> Profile::CopySettings(winrt::com_ptr<Profile> source)
     profile->_Padding = source->_Padding;
     profile->_Commandline = source->_Commandline;
     profile->_StartingDirectory = source->_StartingDirectory;
-    profile->_BackgroundImagePath = source->_BackgroundImagePath;
-    profile->_BackgroundImageOpacity = source->_BackgroundImageOpacity;
-    profile->_BackgroundImageStretchMode = source->_BackgroundImageStretchMode;
     profile->_AntialiasingMode = source->_AntialiasingMode;
-    profile->_RetroTerminalEffect = source->_RetroTerminalEffect;
     profile->_ForceFullRepaintRendering = source->_ForceFullRepaintRendering;
     profile->_SoftwareRendering = source->_SoftwareRendering;
-    profile->_ColorSchemeName = source->_ColorSchemeName;
-    profile->_Foreground = source->_Foreground;
-    profile->_Background = source->_Background;
-    profile->_SelectionBackground = source->_SelectionBackground;
-    profile->_CursorColor = source->_CursorColor;
     profile->_HistorySize = source->_HistorySize;
     profile->_SnapOnInput = source->_SnapOnInput;
     profile->_AltGrAliasing = source->_AltGrAliasing;
-    profile->_CursorShape = source->_CursorShape;
-    profile->_CursorHeight = source->_CursorHeight;
     profile->_BellStyle = source->_BellStyle;
-    profile->_PixelShaderPath = source->_PixelShaderPath;
-    profile->_BackgroundImageAlignment = source->_BackgroundImageAlignment;
     profile->_ConnectionType = source->_ConnectionType;
     profile->_Origin = source->_Origin;
 
+    // Copy over the appearance
+    winrt::com_ptr<AppearanceConfig> sourceDefaultAppearanceImpl;
+    sourceDefaultAppearanceImpl.copy_from(winrt::get_self<AppearanceConfig>(source->_DefaultAppearance));
+    profile->_DefaultAppearance = *AppearanceConfig::CopyAppearance(sourceDefaultAppearanceImpl);
+
+    if (source->_UnfocusedAppearance.has_value())
+    {
+        if (source->_UnfocusedAppearance.value() == nullptr)
+        {
+            profile->_UnfocusedAppearance = nullptr;
+        }
+        else
+        {
+            // Copy over the unfocused appearance
+            winrt::com_ptr<AppearanceConfig> sourceUnfocusedAppearanceImpl;
+            sourceUnfocusedAppearanceImpl.copy_from(winrt::get_self<AppearanceConfig>(source->_UnfocusedAppearance.value()));
+            profile->_UnfocusedAppearance = *AppearanceConfig::CopyAppearance(sourceUnfocusedAppearanceImpl);
+        }

```

Comment:
But then we'll be doing 

`sourceUnfocusedAppearanceImpl.copy_from(winrt::get_self<AppearanceConfig>(source->_UnfocusedAppearance.value()));`

on a potentially `nullptr` value right? Since the `CopyAppearance` call only happens after that line

Reply:
We can still move common logic out of the conditional.

Reply:
Simplified!